packages/grafana-ui/src/components/ConfirmButton/ConfirmButton.tsx

```diff
@@ -0,0 +1,164 @@
+import React, { PureComponent, SyntheticEvent } from 'react';
+import { cx, css } from 'emotion';
+import { stylesFactory, withTheme } from '../../themes';
+import { GrafanaTheme } from '@grafana/data';
+import { Themeable } from '../../types';
+import { Button } from '../Button/Button';
+import { ButtonVariant } from '../Button/types';
+
+const getStyles = stylesFactory((theme: GrafanaTheme) => {
+  return {
+    buttonContainer: css`
+      width: 24px;
+      direction: rtl;
+      display: flex;
+      align-items: center;
+    `,
+    button: css`
+      position: absolute;
+      text-decoration: underline;
+      color: ${theme.colors.linkExternal};
+    `,
+    buttonDisabled: css`
+      text-decoration: none;
+      color: ${theme.colors.text};
+      opacity: 0.65;
+      cursor: not-allowed;
+      pointer-events: none;
+    `,
+    buttonShow: css`
+      opacity: 1;
+      transition: opacity 0.1s ease;
+      z-index: 2;
+    `,
+    buttonHide: css`
+      opacity: 0;
+      transition: opacity 0.1s ease;
+      z-index: 0;
+    `,
+    confirmButtonContainer: css`
+      overflow: hidden;
+      position: absolute;
+      z-index: 1;
+    `,
+    confirmButton: css`
+      display: flex;
+      align-items: flex-start;
+    `,
+    confirmButtonShow: css`
+      opacity: 1;
+      transition: opacity 0.08s ease-out, transform 0.1s ease-out;
+      transform: translateX(0);
+    `,
+    confirmButtonHide: css`
+      opacity: 0;
+      transition: opacity 0.12s ease-in, transform 0.14s ease-in;
+      transform: translateX(100px);
+    `,
+  };
+});
+
+interface Props extends Themeable {
+  className?: string;
+  style?: React.CSSProperties;
+  buttonText?: string;
+  confirmText?: string;
+  disabled?: boolean;
+  confirmButtonVariant?: ButtonVariant;
+  confirmWidth?: number;
+
+  onConfirm(): void;
+  onClick?(): void;
+  onCancel?(): void;
+}
+
+interface State {
+  showConfirm: boolean;
+}
+
+class UnThemedConfirmButton extends PureComponent<Props, State> {
```

Comment:
I assume you use class component here to introduce the simple binary state. I think such cases are perfect fit for using `useState` hook instead.  It would simplify the theming as well, as you could use `useTheme` hook instead of wrapping this component in `withTheme` HOC.

@hugohaggmark wdyt about this case and introducing hooks?

Reply:
@dprokop I've a troubled relationship with hooks so far, they tend to grow into dragons after a while. We need better guidelines for when to use them and when not imho. This could probably be a good candidate though.

Reply:
@dprokop I'm quite new in hooks and theming in grafana, but when I tried to use hooks here, there was an error (looks like theme hook works only with functional components).

Reply:
yes, hooks are functional components only